components/passbook/index.js

```diff
@@ -0,0 +1,63 @@
+import { Component } from 'react';
+import InfiniteScroll from 'react-infinite-scroll-component';
+import styles from './passbook.module.css';
+
+export default class TransactionHistory extends Component {
+  constructor(props) {
+    super(props);
+    let sortedTransactions = [].concat(this.props.transactions);
+
+    sortedTransactions.map((transaction) => {
+      const timeStamp = new Date(transaction.date);
+      transaction.timeStamp = timeStamp.getTime();
+    });
+
+    sortedTransactions.sort((a, b) => b.timeStamp - a.timeStamp);
+    let length = 6;
+    this.state = {
+      allTransaction: Array.from(sortedTransactions.slice(0, length)),
+      hasMore: true,
+    };
+
+    this.fetchMoreData = () => {
+      if (this.state.allTransaction.length >= sortedTransactions.length) {
+        this.setState({ hasMore: false });
+        return;
+      }
+      length += 2;
+      setTimeout(() => {
+        this.setState({
+          allTransaction: sortedTransactions.slice(0, length),
+        });
+      }, 500);
+    };
+  }
+  render() {
+    const showThem = this.state.allTransaction.map((transaction) => {
+      return (
+        <div className={styles.transaction} key={transaction.timeStamp}>
```

Comment:
Any specific reason why you are using `key` as the timestamp?

Reply:
As the `timestamp` contains the time and date of transaction accurately till the order of milliseconds,  it should stay unique for the most part and hence could be used as key. Feel free to correct me.

Reply:
you can also use the index as the key, just in case if you would like to consider this. The anonymous function that gets called in the `map` function , it also receives the second argument as `index`
so your code can be `this.state.allTransaction.map((transaction,index) => {) `
use this `index` as your key